packages/synthetic-chain/public/upgrade-test-scripts/env_setup.sh

```diff
@@ -60,12 +60,31 @@ if [[ "$binary" == "agd" ]]; then
   sed -i 's/minSubmissionCount": 3/minSubmissionCount": 1/g' "$configdir"/*.json
 fi
 
+# XXX race conditions between build stages
+await_agd_startable() {
+  local retries="$1"
+
+  local wait="10s"
+
+  if agd status >/dev/null 2>&1; then
+    # agd is running
+    if [[ $retries -gt 0 ]]; then
+      echo "Waiting $wait for agd to stop"
+      sleep $wait
+
+      wait_and_retry $(($retries - 1))
+    else
+      echo "Cannot start agd because it's already running"
+      return 1
+    fi
+  fi
+}
+
 startAgd() {
   echo "startAgd()"
 
-  # XXX debugging https://github.com/Agoric/agoric-3-proposals/issues/93
-  apt install net-tools
-  netstat
+  # precondition check
```

Comment:
https://github.com/Agoric/agoric-3-proposals/pull/127#issuecomment-1998111339
```suggestion
  # XXX debugging https://github.com/Agoric/agoric-3-proposals/issues/93
  apt install net-tools
  netstat -plnA inet
  # precondition check
```

Reply:
This PR is to remove the netstat debugging. The underlying cause is that `agd` is already bound to the ports. I don't know the cause of that but this hacks around it by waiting for it to die.

Reply:
Right, I'm trying to get to the bottom of that with a more useful invocation of `netstat`.

Reply:
Oh, I thought you were suggesting a change in this PR. If you want to figure out the cause of failures in https://github.com/Agoric/agoric-3-proposals/pull/127 I thought you'd push to that branch whatever logging you need.

Reply:
I'm putting this up for review again because I think we should keep those netstat logs out of main. They make the logs unwieldy to scroll in the Github UI